src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -291,31 +292,166 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(PhysicalLocation v2Ph
             return physicalLocation;
         }
 
-        internal RegionVersionOne CreateRegion(Region v2Region)
+        internal RegionVersionOne CreateRegion(Region v2Region, Uri uri)
         {
             RegionVersionOne region = null;
 
             if (v2Region != null && (v2Region.StartColumn > 0 ||
                                      v2Region.StartLine > 0 ||
                                      v2Region.EndColumn > 0 ||
                                      v2Region.EndLine > 0 ||
-                                     v2Region.Length > 0 ||
-                                     v2Region.Offset > 0))
+                                     v2Region.ByteLength > 0 ||
+                                     v2Region.ByteOffset > 0 ||
```

Comment:
This was a bug in the original code. `ByteOffset` can validly be 0 if the region starts at the beginning of the file. `ByteLength` can be 0 if the region represents an insertion point.  Similarly for the `Char` properties.

So... the original logic, which attempted to use the property values to decide whether to create a region, was just wrong. If there is a `v2Region` object, you have to convert it, even if all its properties are 0 -- that just means "an insertion point at the beginning of the file".

So this should just be:
```
if (v2Region != null)
{
    region = new RegionVersionOne()
    {
        ...
    };
}
```

Reply:
Good, fixed.